packages/adblocker/src/engine/engine.ts

```diff
@@ -326,6 +349,22 @@ export default class FilterEngine extends EventEmitter<
     // Subscription management: disabled by default
     this.lists = lists;
 
+    // Preprocessors
+    this.preprocessors = new PreprocessorBucket({ env, preprocessors });
+
+    // A hack to pass the serialization test calling diff on the `FiltersEngine`
+    // Instead of overriding the prototype of `FiltersEngine.prototype.isFiltersExcluded`,
+    // we go with Object.defineProperty.
+    // If you're going to improve this part in future, playing with `deep-object-diff`
+    // can help you.
+    const isFilterExcluded = (filter: IFilter) => {
+      return this.preprocessors.isFilterExcluded(filter);
+    };
+
+    Object.defineProperty(this, 'isFilterExcluded', {
+      value: isFilterExcluded,
+    });
```

Comment:
I think this can be replaced by this method on FiltersEngine class itself (using the arrow syntax combined with class method syntax):
```typescript
private isFilterExcluded = (filter: IFilter) => {
  return this.preprocessors.isFilterExcluded(filter);
}
```

Reply:
I can't say the exact reason why direct lambda property didn't work but it didn't work. You may try with `cd packages/adblocker && yarn test ./test/serialization.test.ts`. I assume it's because it broke the prototype chain which is checked in diffing function.

Reply:
Resolved by [`2084cdd` (#3782)](https://github.com/ghostery/adblocker/pull/3782/commits/2084cdde34d033de4245b8a9226c4796ce5c6bc6)